sdk/tables/Azure.Data.Tables/src/TableServiceClient.cs

```diff
@@ -146,14 +146,14 @@ public TableServiceClient(Uri endpoint, TableSharedKeyCredential credential, Tab
         /// <param name="options">
         /// Optional client options that define the transport pipeline policies for authentication, retries, etc., that are applied to every request.
         /// </param>
-        public TableServiceClient(string connectionString, TableClientOptions options = null)
+        public TableServiceClient(string connectionString, TablesClientOptions options = null)
         {
             Argument.AssertNotNull(connectionString, nameof(connectionString));
 
             TableConnectionString connString = TableConnectionString.Parse(connectionString);
             _accountName = connString._accountName;
 
-            options ??= new TableClientOptions();
+            options ??= new TablesClientOptions();
```

Comment:
Are we concerned with options being mutated after construction?   Should we consider a defensive copy?